### ISSUE TITLE
Any examples or guidance for using CRDs with the js api?

### ISSUE THREAD

**Issue by weikinhuang** [kubernetes-client/javascript#144]
There's no examples on how to extend the client library to be able to use CRDs via the same api interface as the rest of the kubernetes api.

For example the godaddy/kubernetes-client lets you extend their api manually, see https://github.com/godaddy/kubernetes-client/blob/master/examples/using-crds.js

**Comment by flyboarder** [comment]
All the API's work the same, see #117

`.createCustomResourceDefinition()`

**Comment by weikinhuang** [comment]
Sorry about the confusion, but i'm more talking about crud operations on the extended apis created by CRDs not the CRD itself. Since CRDs are part of the api, and the resources created by the crds are not.

**Comment by NicolaiSchmid** [comment]
@brendandburns Could you provide a little bit more context for me?  
I'm trying to use the custom `kubervirt.io` API and I'm completely lost, how to use `createClusterCustomObject` to create the CRD in the client and then query for the `virtualmachine` resource.
The CRD itself looks like this: https://pastebin.com/jvgbfbJA

**Comment by flyboarder** [comment]
Something like this:

```js
...
var k8sAppApiClient = kc.makeApiClient(k8s.Apps_v1beta1Api);
...

k8sAppApiClient. createClusterCustomObject(....);
...
```

**Comment by xujihui1985** [comment]
@brendandburns I don't know why this issue is closed, and I don't see an answer here, How can I apply an CR with this client?

**Comment by arianbessonart** [comment]
I'm trying to retrieve `certificates` from `certmanager.k8s.io/v1alpha1` group/version. 
The certificate object I already created running `kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.8.0/cert-manager.yaml
` in the GCP console.

How can I use the client to retrieve `certificates` as well as `issuers`?

**Comment by xujihui1985** [comment]
@ankitabhopatkar13 
const client = kc.makeApiClient(k8s.Custom_objectsApi)
await client.getNamespacedCustomObject(
  'certmanager.k8s.io',
  'v1alpha1',
  '<your namespace>',
  '<your resource name>',
)

**Comment by ankitabhopatkar13** [comment]
@xujihui1985 I think you wanted to address @arianbessonart in above comment ?

**Comment by xujihui1985** [comment]
@ankitabhopatkar13 yes, sorry I'm blind 😹

**Comment by arianbessonart** [comment]
@xujihui1985 thanks!!!
This is very helpful.

**Comment by fentas** [comment]
Still, a bit confused here.
`k8s.Custom_objectsApi` is undefined for me, do I have to define this somewhere?

```
/application/node_modules/@kubernetes/client-node/dist/config.js:184
        const apiClient = new apiClientType(cluster.server);
                          ^

TypeError: apiClientType is not a constructor
```

**Comment by xujihui1985** [comment]
@fentas no, you don't have to, Custom_objectsApi is a built in Class, do some log before `makeApiClient`  call to see what you really passed

**Comment by fentas** [comment]
```js
const k8s = require('@kubernetes/client-node')
const kc = new k8s.KubeConfig()
kc.loadFromCluster()
// this is undefined
onsole.log(k8s.Custom_objectsApi)
// this will throw error above
const client = kc.makeApiClient(k8s.Custom_objectsApi)
```

using `"@kubernetes/client-node": "^0.10.1"`

**Comment by hcharley** [comment]
Here is some code I wrote that gets a single instance of a CRD:

https://gist.github.com/timmyers/4d2fed53a358d4c98557a5886ae2afbb?permalink_comment_id=4153572#gistcomment-4153572
